tools/xkbcli-completion.sh

```diff
@@ -0,0 +1,97 @@
+# bash completion support for xkbcli.
+
+# See completion API documentation: https://github.com/scop/bash-completion
+# NOTE: The script parses the commands help messages to provide the completions,
+#       thus any new subcommand or option will be supported, as long as it has its
+#       entry in the help messages. This should result in low maintenancei effort.
+
+___xkbcli_main()
+{
+    # Initialization: https://github.com/scop/bash-completion/blob/fdf4456186eb4548ef628e65fb1be73d8e4695e9/bash_completion.d/000_bash_completion_compat.bash#L205
+    local cur prev words cword
+    _init_completion -s || return
+
+    local i=1 cmd
+
+    # Find subcommand
+    while [[ "$i" -lt "$COMP_CWORD" ]]
+    do
+        local s="${COMP_WORDS[i]}"
+        case "$s" in
+            -*) ;;
+            *)
+            cmd="$s"
+            break
+            ;;
+        esac
+        (( i++ ))
+    done
+
+    # Parse available subcommands
+    local help_output line is_command_list=false subcommands=()
+    help_output=$(xkbcli --help)
+    while IFS= read -r line; do
```

Comment:
```suggestion
    while IFS='' read -r line; do
```
for clarity? or is this not the same? my bash scripting is quite rusty